designdocs/schemas_protocols_intents.md

```diff
@@ -0,0 +1,642 @@
+
+- **protocol** - the top level of the tripartite nomenclature _protocol.schema@version_ (ex: _dsnp_ is the protocol in
+  _dsnp.broadcast@v2_)
+- **schema** - the second level of _protocol.schema@version_ (ex: _dsnp.broadcast_ resolves to a specific schema).
+  Schemas should always be referred to by their fully-qualfied name (ie, _dsnp.broadcast_, not just _broadcast_)
+- **version** - the third level of _protocol.schema@version_, resolves to a specific _minor_ iteration of a schema
+- **intent** - an on-chain primitive representing an abstract action or operation
+
+## 4. **Schemas and Versioning** <a id="section_4"></a>
+
+Schemas are not just organizational tools—they enable meaningful delegation. In the new model, a schema is a
+first-class on-chain entity that represents a named and versioned group of _schema versions_. Every schema version is
+assigned to a
+schema, and new versions of a schema simply add new schema version IDs to the schema's version history.
+
+### 📌 Schema Metadata Structure
+
+- `SchemaId`: A unique numeric identifier for runtime efficiency
+- `ProtocolId`: Which protocol owns this schema
+- `Name`: Human-readable schema name (e.g., `broadcast`)
+- `Versions`: Ordered list of `SchemaVersionId`s (e.g., v0 → v1 → v2)
+- Metadata: Optional description, links, etc.
+
+### 🧬 Versioning Rules
+
+- Versions are stored in a monotonic, gapless array
+- A new schema version ID is automatically assigned the next version
+- Older versions are preserved permanently
+- Delegations may apply to the schema, not a specific schema version ID
+- _Question: can versions be deprecated?_
+
+### 🧭 Governance Rules
+
+- New schemas must be approved by governance
+- Minor updates (new schema version) may be published by the protocol owner
+- Major updates require a new schema (e.g., change in semantic intent)
+
+## 5. **Protocol Ownership** <a id="section_5"></a>
+
+In order to provide organizational control, accountability, and publishing boundaries for schemas and intents, the
+protocol introduces the concept of **protocols**.
+
+A protocol (e.g., `dsnp`, `bsky`) serves as a root authority under which schemas and intents are registered.
+Protocols allow developers and governance bodies to clearly define who may publish what, and to prevent protocol
+squatting or unauthorized use of shared protocol identifiers.
+
+### 📛 Protocol Structure
+
+Each protocol is a unique identifier, typically human-readable (e.g., `"dsnp"`), mapped to an on-chain `ProtocolId` (
+e.g., `ProtocolId = 2`).
+
+- Every schema must be published under a protocol: `protocol.schema`
+- Intents may also be published under a protocol _(open question, discussed [below](#section_7))_
+
+### 👤 Ownership and Control
+
+Protocols are envisioned to be owned by an entity, but there is an open question as to whether this entity should be an
+MSA (i.e., a Provider) or a raw account (i.e., a public key). This decision will impact how access is managed,
```

Comment:
One of the advantages of MSAs is that, in theory, we can attach permissions to them. This would allow governance to add keys with limited access, which is especially valuable for organizations that need to delegate responsibilities. Being able to grant keys with restricted functionality becomes even more important in these contexts.

Reply:
Interesting. Probably the subject of a separate design or issue, but the idea of adding a control key with explicit/limited rights is intriguing... though, delegations could accomplish the same thing.